packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -991,41 +1138,69 @@ export class IntervalCollection<TInterval extends ISerializableInterval>
 	 * {@inheritdoc IIntervalCollection.add}
 	 */
 	public add(
-		start: number,
-		end: number,
+		start: number | "start" | "end" | SequencePlace,
+		end: number | "start" | "end" | SequencePlace,
 		intervalType: IntervalType,
 		props?: PropertySet,
-		stickiness: IntervalStickiness = IntervalStickiness.END,
 	): TInterval {
 		if (!this.localCollection) {
 			throw new LoggingError("attach must be called prior to adding intervals");
 		}
 		if (intervalType & IntervalType.Transient) {
 			throw new LoggingError("Can not add transient intervals");
 		}
-		if (stickiness !== IntervalStickiness.END && !this.options.intervalStickinessEnabled) {
+
+		const { startSide, endSide, startPos, endPos } = endpointPosAndSide(start, end);
+
+		assert(
+			startPos !== undefined &&
+				endPos !== undefined &&
+				startSide !== undefined &&
+				endSide !== undefined,
+			"start and end cannot be undefined because they were not passed in as undefined",
+		);
+
+		let stickiness: IntervalStickiness = IntervalStickiness.NONE;
+
+		if (startSide === Side.After || start === "start" || start === "end") {
```

Comment:
I think this should be
```suggestion
		if (startSide === Side.After || start === "start") {
```

and `end === "start"` removed from the one below (L1169)